model/config.go

```diff
@@ -2248,6 +2248,7 @@ type PluginSettings struct {
 	RequirePluginSignature   *bool
 	MarketplaceUrl           *string
 	SignaturePublicKeyFiles  []string
+	CanaryTag                *string
```

Comment:
If I may chime in: From what I understand the main way of using `CanaryTag` is via environment variables. If a server wants the "default" behaviour, he would just not set `CanaryTag`.

What is the use case of a `CanaryTag` in the config file?

Reply:
This exposes the canary tag to the system console in the standard way, even though it will be read-only when set. The idea is to use the existing `MM_` paradigm and expose visibility to the current configuration in the system console, vs. having to honour a "special" environment variable and find a way to expose the current one somewhere. Hackathon-focussed, primarily.

Reply:
Thanks for the explanation. So there should not be a support for setting this config value in the config file/db?

Reply:
I don't think there's a way to get all the system console / `MM_` integration /without/ also making it a configuration bit. We could replicate all of that in a bid to try to keep it out of the config struct, but it seems heavyweight for something experimental.

Reply:
Hm, I'm still wondering why this setting needs to be shown in the system console. Just allowing it to be set via an environment variables and not including it into the config struct seams easier to me.

Reply:
I'm hesitant to allow `MM_*` overrides to be used for something not in the config struct -- it seems like there's benefits to this being consistent here, especially when we can query a server to determine it's canary tag programmatically. Showing it in the system console is just an extension to that for debugging purposes.

Reply:
Fair point. I'm at most 1/5, so the implementation is fine.